File: lectures/code_snippets/fixture_test.py

import pytest
import numpy as np

class Particle:
    def __init__(self):
        position = np.zeros(3)
        velocity = np.zeros(3)
        force = np.zeros(3)

class System:
  particles = []

@pytest.fixture
def SystemTest():
    # SetUp
    System.particles = [Particle() for i in range(0, 10)]
    yield 
    # TearDown
    System.particles = []

def test_is_not_empty(SystemTest):
    assert len(System.particles) == 10  

File: lectures/code_snippets/fixture_test.cc
#include <algorithm>
#include <cmath>
#include <exception>
#include <gtest/gtest.h>
#include <numeric>
#include <vector>

struct Particle {

  double position[3];
  double velocity[3];
  double force[3];
};

struct System {
  std::vector<Particle> particles;
};

// Fixture class
class SystemTest : public ::testing::Test {
protected:
  void SetUp() override {
    // push 10 particles
    system.particles.resize(10);
  }

  void TearDown() override {
    // clear particles
    system.particles.clear();
  }
  System system;
};

// Fixture test
TEST_F(SystemTest, IsNotEmpty) {
  // check the size of container
  EXPECT_EQ(system.particles.size(), 10);
}